csi/node.go

```diff
@@ -113,6 +114,15 @@ func (s *OsdCsiServer) NodePublishVolume(
 	// Get volume encryption info from req.Secrets
 	driverOpts := s.addEncryptionInfoToLabels(make(map[string]string), req.GetSecrets())
 
+	// Parse storage class 'mountOptions' flags from CSI req
+	// flags from 'mountOptions' will be used as the only source of truth for Pure volumes upon mounting
+	if req.GetVolumeCapability() != nil && req.GetVolumeCapability().GetMount() != nil {
+		mountFlags := strings.Join(req.GetVolumeCapability().GetMount().GetMountFlags(), ",")
+		if mountFlags != "" {
+			driverOpts[api.SpecMountOptions] = mountFlags
```

Comment:
Let's create a new `api.SpecCSIMountOptions = csi_mount_options`, separate from the standard Portworx mount options we support. 

I don't want to conflict with any existing Portworx mount options that can be passed in via storageclass parameters. This might not affect Pure integration, but if we want to support csi_mount_options in Portworx in the future, it will make things easier.

Reply:
Also, let's create a separate `mountOptions` map here and pass that to mounts.Mount below. Let's not re-use the driverOpts used for encryption secrets on attach.

Reply:
Discussed offline, okay to use existing driverOpts with the new csi mount options flag.

Reply:
@Pure-AdamuKaapan this might help answer your question about csi_mount_options